src/transformers/trainer.py

```diff
@@ -1170,6 +1170,38 @@ def get_optimizer_cls_and_kwargs(args: TrainingArguments) -> Tuple[Any, Any]:
                 optimizer_kwargs.update(adam_kwargs)
             except ImportError:
                 raise ValueError("Trainer tried to instantiate apex FusedAdam but apex is not installed!")
+        elif args.optim in [
```

Comment:
The optimizer should go in their own PRs, let's keep this one focused on one thing.

Reply:
These changes would also be needed. Can you please have a look at: https://github.com/huggingface/transformers/pull/23217